src/v/datalake/record_schema_resolver.cc

```diff
@@ -473,7 +473,10 @@ latest_subject_schema_resolver::resolve_buf_type(std::optional<iobuf> b) const {
       = co_await ss::coroutine::as_future<ppsr::stored_schema>(
         sr_->get_subject_schema(subject_, /*subject_version=*/std::nullopt));
     if (latest_schema_fut.failed()) {
-        latest_schema_fut.ignore_ready_future();
+        vlog(
+          datalake_log.warn,
+          "Error getting subject schema from registry: {}",
+          latest_schema_fut.get_exception());
```

Comment:
@nvartolomei can you pull `get_exception()` out of the `vlog` statement? we want to keep logging statements free of side-effects, and get_exception does have a side-effect: it marks the future as consumed so that it doesn't throw abandonded exceptional future